crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2209,8 +2209,25 @@ impl<'db> TypeInferenceBuilder<'db> {
         } = unary;
 
         match (op, self.infer_expression(operand)) {
+            (UnaryOp::UAdd, Type::IntLiteral(value)) => Type::IntLiteral(value),
             (UnaryOp::USub, Type::IntLiteral(value)) => Type::IntLiteral(-value),
+            (UnaryOp::Invert, Type::IntLiteral(value)) => Type::IntLiteral(-(value + 1)),
+
+            (UnaryOp::UAdd, Type::BooleanLiteral(bool)) => {
+                let value = i64::from(bool);
+                Type::IntLiteral(value)
+            }
+            (UnaryOp::USub, Type::BooleanLiteral(bool)) => {
+                let value = i64::from(bool);
+                Type::IntLiteral(-value)
+            }
+            (UnaryOp::Invert, Type::BooleanLiteral(bool)) => {
+                let value = i64::from(bool);
+                Type::IntLiteral(-(value + 1))
+            }
+
             (UnaryOp::Not, ty) => ty.bool(self.db).negate().into_type(self.db),
+
             _ => Type::Unknown, // TODO other unary op types
```

Comment:
nit: I'd probably just inline these

```suggestion
            (UnaryOp::UAdd, Type::BooleanLiteral(bool)) => Type::IntLiteral(i64::from(bool)),
            (UnaryOp::USub, Type::BooleanLiteral(bool)) => Type::IntLiteral(-i64::from(bool))
            (UnaryOp::Invert, Type::BooleanLiteral(bool)) => Type::IntLiteral(!i64::from(bool))
            (UnaryOp::Not, ty) => ty.bool(self.db).negate().into_type(self.db),
            _ => Type::Unknown, // TODO other unary op types
```